File: cross_lang/numeric/mandelbrot.js
function mandelbrot_point(cr, ci, max_iter) {
    let zr = 0.0;
    let zi = 0.0;
    let iter = 0;
    while (iter < max_iter) {
        let zr2 = zr * zr;
        let zi2 = zi * zi;
        if (zr2 + zi2 > 4.0) {
            return iter;
        }
        zi = 2.0 * zr * zi + ci;
        zr = zr2 - zi2 + cr;
        iter++;
    }
    return max_iter;
}

function compute_grid(width, height, max_iter) {
    let total = 0;
    for (let y = 0; y < height; y++) {
        let ci = -1.5 + (3.0 * y) / height;
        for (let x = 0; x < width; x++) {
            let cr = -2.0 + (3.0 * x) / width;
            let iters = mandelbrot_point(cr, ci, max_iter);
            total += iters;
        }
    }
    return total;
}

const total = compute_grid(200, 100, 200);
console.log(total);

File: cross_lang/numeric/mandelbrot.c
#include <stdio.h>

int mandelbrot_point(double cr, double ci, int max_iter) {
    double zr = 0.0;
    double zi = 0.0;
    int iter = 0;
    while (iter < max_iter) {
        double zr2 = zr * zr;
        double zi2 = zi * zi;
        if (zr2 + zi2 > 4.0) {
            return iter;
        }
        zi = 2.0 * zr * zi + ci;
        zr = zr2 - zi2 + cr;
        iter++;
    }
    return max_iter;
}

long long compute_grid(int width, int height, int max_iter) {
    long long total = 0;
    for (int y = 0; y < height; y++) {
        double ci = -1.5 + (3.0 * y) / height;
        for (int x = 0; x < width; x++) {
            double cr = -2.0 + (3.0 * x) / width;
            int iters = mandelbrot_point(cr, ci, max_iter);
            total += iters;
        }
    }
    return total;
}

int main() {
    long long total = compute_grid(200, 100, 200);
    printf("%lld\n", total);
    return 0;
}
